scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+            pubKeyPath = cmd.getOptionValue(KEYFILE_OPTION_NAME);
+        }
+
+        if (pubKeyPath == null) {
+            pubKeyPath = getPublicKeyFilePath();
+        }
+        try {
+            pubKey = Credentials.getPublicKey(pubKeyPath);
+        } catch (KeyException e) {
+            exitWithErrorMessage("Could not retrieve public key from '" + pubKeyPath, null, e);
+        }
+
+        switch (action) {
+            case CREATE:
+                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null && userInfo.getGroups() != null);
+                break;
+            case UPDATE:
+                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null) || (userInfo.getLogin() != null && userInfo.getGroups() != null);
+                break;
+            case DELETE:
+                nonInteractive = userInfo.getLogin() != null;
+                break;
+        }
+
+        if (!nonInteractive) {
+            askInteractively(console, userInfo, action);
+        }
+
+        updateAccounts(pubKey, userInfo, loginFilePath, groupFilePath, action);
+    }
+
+    /**
+     * Ask the user for parameters missing on the command line
+     */
+    private static void askInteractively(Console console, UserInfo userInfo, Action action) {
+        System.out.println(action);
+        System.out.print("login: ");
+        if (userInfo.getLogin() == null) {
+            userInfo.setLogin(console.readLine());
+        } else {
+            System.out.println(userInfo.getLogin());
+        }
+        System.out.print("password: ");
+        if ((action.isCreate() && userInfo.getPassword() == null) || (action.isUpdate() && userInfo.getPassword() == null && userInfo.getGroups() == null)) {
+            userInfo.setPassword(new String(console.readPassword()));
+        } else {
+            System.out.println("*******");
+        }
+
+        if (action.isCreate() && userInfo.getGroups() == null) {
+            System.out.print("groups for user " + userInfo.getLogin() + ":");
+            String groupString = console.readLine();
+            userInfo.setGroups(groupString.split(","));
+        }
+    }
+
+    /**
+     * Update the accounts in the login and config files
+     *
+     * @throws ManageUsersException
+     */
+    private static void updateAccounts(PublicKey pubKey, UserInfo userInfo, String loginFilePath, String groupFilePath, Action action) throws ManageUsersException {
+        try {
+            Properties props = new Properties();
+            try {
+                FileInputStream stream = new FileInputStream(new File(loginFilePath));
+                props.load(stream);
+                stream.close();
+            } catch (Exception e) {
+                exitWithErrorMessage("could not read login file : " + loginFilePath, null, e);
+            }
+
+            Multimap<String, String> groupsMap = loadGroups(groupFilePath);
+
+            switch (action) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)